utils/camping_argparser.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import logging
+import sys
+
+from datetime import datetime
+
+from enums.date_format import DateFormat
+
+
+class CampingArgParser(argparse.ArgumentParser):
+
+    def __init__(self):
+        super().__init__()
+        self.add_argument("--debug", "-d", action="store_true", help="Debug log level")
+        self.add_argument(
+            "--start-date", required=True, help="Start date [YYYY-MM-DD]", type=self.TypeConverter.date
+        )
+        self.add_argument(
+            "--end-date",
+            required=True,
+            help="End date [YYYY-MM-DD]. You expect to leave this day, not stay the night.",
+            type=self.TypeConverter.date,
+        )
+        self.add_argument(
+            "--nights",
+            help="Number of consecutive nights (default is all nights in the given range).",
+            type=self.TypeConverter.positive_int,
+        )
+        self.add_argument(
+            "--campsite-ids",
+            type=int,
+            nargs="+",
+            default=(),
+            help="Optional, search for availability for a specific campsite ID.",
+        )
+        self.add_argument(
+            "--show-campsite-info",
+            action='store_true',
+            help="Display campsite ID and availability dates.",
+        )
+        self.add_argument(
+            "--campsite-type",
+            help=(
+                "Optional, can specify type of campsite such as:"
+                '"STANDARD NONELECTRIC" or TODO'
+            ),
+        )
+        self.add_argument(
+            "--json-output",
+            action="store_true",
+            help=(
+                "This make the script output JSON instead of human readable "
+                "output. Note, this is incompatible with the twitter notifier. "
+                "This output includes more precise information, such as the exact "
+                "avaiable dates and which sites are available."
+            ),
+        )
+        parks_group = self.add_mutually_exclusive_group(required=True)
+        parks_group.add_argument(
+            "--parks", dest="parks", metavar="park", nargs="+", help="Park ID(s)", type=int
+        )
+        parks_group.add_argument(
+            "--stdin",
+            "-",
+            action="store_true",
+            help="Read list of park ID(s) from stdin instead",
+        )
+
+    def parse_args(self, args=None, namespace=None):
+        args = super().parse_args(args, namespace)
+        args.parks = args.parks or [p.strip() for p in sys.stdin]
+        self._validate_args(args)
+        return args
+
+    @staticmethod
```

Comment:
Same comment as before, and could you apply it generally? If you have a method of a class that refers to the class, make it a class method.

Reply:
Done.